### PR TITLE
Release candidate testing, and release specific changes

### PR DESCRIPTION
**Pull request by wasade**
Provides support for #743 

This PR is also scoped now to include items on #741

### CODE REVIEW

**Comment by wasade** [comment]
@ElDeveloper @antgonza @josenavas Possible for review? This PR catches a few items which were caught on release testing, version/copyright/etc updates, and scripts which exercise both the command line and API that were developed to facilitate release testing. It is large but I do not think this will be a complex review, and many of the LOC are due to regeneration of the Cython .c files.

Once this PR is merged, we can transition to the remaining items in #741.

File: biom/assets/exercise_api.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2011-2015, The BIOM Format Development Team.
```
Comment (josenavas) [inline biom/assets/exercise_api.py]: 2015 -> 2017

Probably better to use 'current' or leave it open so we don't have to change every year?

Comment (josenavas) [inline biom/assets/exercise_api.py]: @wasade what is the goal of this file? I understand that it is exercising the API, but given that those calls are part of the Table object, shouldn't those functions be tested using the unit-tests of the Table object? Or am I missing something?

**Comment by wasade** [comment]
Thanks!

The exercise API script is a set of integration tests. The unit tests of
course are in test_table.py

No strong feelings on year or current

**Comment by wasade** [comment]
C files are auto generated

**Pull request merged**